app/src/androidTestWebview/java/org/mozilla/focus/activity/SearchEngineSelectionTest.java

```diff
@@ -60,85 +74,140 @@ public void tearDown() throws Exception {
+
+
+        // Verify new engine default
+        {
+            TestHelper.pressBackKey();
+            TestHelper.settingsHeading.waitForExists(waitingTime);
+
+            UiObject defaultSearchEngine = TestHelper.mDevice.findObject(new UiSelector()
+                    .text(mSearchEngine)
+                    .resourceId("android:id/title"));
+
+            assertTrue(defaultSearchEngine.getText().equals(mSearchEngine));
+            TestHelper.pressBackKey();
+            TestHelper.pressBackKey();
+        }
+
+
+        // Do search on blank spaces and press enter for search (should not do anything)
+        {
+            TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+            TestHelper.inlineAutocompleteEditText.clearTextField();
+            TestHelper.inlineAutocompleteEditText.setText("   ");
+            TestHelper.pressEnterKey();
+            assertTrue(TestHelper.inlineAutocompleteEditText.exists());
+        }
+
+
+        // Do search on text string
+        {
+            TestHelper.inlineAutocompleteEditText.clearTextField();
+            TestHelper.inlineAutocompleteEditText.setText("mozilla focus");
+            TestHelper.hint.waitForExists(waitingTime);
+        }
+
+
+        // Check that search hint bar correctly displayed
+        {
+            assertTrue(TestHelper.hint.getText().equals("Search for mozilla focus"));
+            TestHelper.hint.click();
+        }
+
+
+        // Verify browser shows google search webview
+        {
+            googleWebView.waitForExists(waitingTime);
+            TestHelper.progressBar.waitForExists(webPageLoadwaitingTime);
+            Assert.assertTrue(TestHelper.progressBar.waitUntilGone(webPageLoadwaitingTime));
+        }
+
+
+        // Search for words: <Google|DuckDuckGo|etc.>, mozilla, focus
+        {
+            assertTrue(TestHelper.browserURLbar.getText().contains(mSearchEngine.toLowerCase()));
+            assertTrue(TestHelper.browserURLbar.getText().contains("mozilla"));
+            assertTrue(TestHelper.browserURLbar.getText().contains("focus"));
+        }
+
+
+        // Tap URL bar, check it displays search term (instead of URL)
+        {
+            TestHelper.browserURLbar.click();
+            TestHelper.inlineAutocompleteEditText.waitForExists(waitingTime);
+            assertEquals(TestHelper.inlineAutocompleteEditText.getText(), "mozilla focus");
+            TestHelper.pressEnterKey();
+            googleWebView.waitForExists(waitingTime);
+            TestHelper.progressBar.waitForExists(webPageLoadwaitingTime);
+            Assert.assertTrue(TestHelper.progressBar.waitUntilGone(webPageLoadwaitingTime));
+            assertTrue(TestHelper.browserURLbar.getText().contains(mSearchEngine.toLowerCase()));
+            assertTrue(TestHelper.browserURLbar.getText().contains("mozilla"));
+            assertTrue(TestHelper.browserURLbar.getText().contains("focus"));
+        }
+
+
+        // Do another search on text string
```

Comment:
@npark-mozilla was the original intent of this to simply repeat the above block to verify consistency?  if so, I'll just duplicate it